.travis.yml

```diff
@@ -1,5 +1,7 @@
 sudo: required
 
+# We need the systemd for the kubeadm and it's default from 16.04+
+dist: xenial
```

Comment:
Is it safe to use `dist: xenial`?

I was researching about it, however, I was not able to find any fresh information about that. The newest information I found is about 3 months old.

16.04 (Xenial) is still not officially supported by Travis-CI. The image is available, but it can disappear at any time and it is not guaranteed that builds will work.

Some references:

* [The Travis-CI documentation](https://docs.travis-ci.com/user/reference/overview/) states only that Trusty, Precise (note that 12.04 has reached EOL) and macOS. However, it doesn't state when the document has been updated last time.
* [travis-ci/travis-ci#5821](https://github.com/travis-ci/travis-ci/issues/5821), the old one, but has some context about the issues. At that time (2016 and 2017), it was not know when is Xenial going to get published.
* [travis-ci/travis-ci#7260](https://github.com/travis-ci/travis-ci/issues/7260). The last comment is from 3-4 months ago, but no update from Travis team, and the issue got locked after that. [Also, there's a comment stating that it may or may not work, and that it's not supported](https://github.com/travis-ci/travis-ci/issues/7260#issuecomment-354350596).
* [This PR (travis-ci/travis-yml#16)](https://github.com/travis-ci/travis-yml/pull/16) added Xenial support to Travis-CI, but marked as Edge feature ([it is still marked as an edge feature](https://github.com/travis-ci/travis-yml/blob/master/spec/travis/yaml/spec/def/dist_spec.rb#L15)). Following the mentioned issues from that one, there are reported problems from April that builds were randomly failing.

However, some of the problems seems solved. The build details are now correct, and reports Xenial instead of Trusty:
```
Operating System Details
Distributor ID:	Ubuntu
Description:	Ubuntu 16.04.5 LTS
Release:	16.04
Codename:	xenial
```

I don't have strong opinions on this one. Moreover, I would really love this to get merged! The `localkube` bootstrap is deprecated by the Minikube team, and switching to `kubeadm` would help a lot.

Reply:
@hwchiu I agree with you.

What we can do is to create a new branch, and then merge the PR against that branch. Then update the README files in all three branches to include a short comparison between each other and note what's the recommended solution.

In that case, we would have examples for all three cases.

I think you can't create a new branch, but if that makes sense and Lili agrees, Lili or I can do it for you. Then, you will be able to update the PR to use the new branch.

Reply:
@hwchiu I created a new branch https://github.com/LiliC/travis-minikube/tree/kubeadm/xenial feel free to change the PR to be merged to that. And we can after that do a followup with the README updated. Thanks! Sorry for not getting back to your earlier!

Reply:
@LiliC 
Thanks your help 👍 
Since the base branch of this PR is different of `kubeadm/xenial` branch, I will rebase this PR to `kubeadm/xenial ` branch and let you know when I done.

Reply:
@hwchiu Looks like you're based on the 1.10 branch. I will recreate the `kubeadm/xenial`, so you can merge without creating a new PR.

Reply:
@hwchiu I've created a new `kubeadm/xenial` branch based of 1.10 branch. Try again now and let me know does it work for you.